### PR TITLE
BUG: support GeoDataFrame with only geometry column and no data

### LINKED ISSUE #559: GeoDataFrame constructor doesn't work on geometry-only DataFrame [cython]
(see http://disq.us/p/1mi03gt)

With geopandas 0.3.0, the following worked in a notebook:

    geoms = [shapely.geometry.Point(1, 1)]
    df = gpd.GeoDataFrame(geometry=geoms)
    geoms

However, in the *geopandas-cython* branch, I get a `ValueError: not enough values to unpack (expected 1, got 0)`.

### PR DESCRIPTION
**Pull request by andreas-h**
fixes #559

### CODE REVIEW

**Comment by andreas-h** [comment]
ah, sorry, mistake. need to think again.

File: geopandas/geodataframe.py [hunk 1]
```diff
@@ -88,6 +88,8 @@ def __init__(self, *args, **kwargs):
 
         if isinstance(arg, (dict, list, pd.Series, np.ndarray)):
             arg = pd.DataFrame(arg, **kwargs)
+        elif arg is None:
+            arg = pd.DataFrame(**kwargs)
```
Comment (jorisvandenbossche) [inline geopandas/geodataframe.py]: this should not be necessary, as `DataFrame(None)` (what you would have in `pd.DataFrame(arg, **kwargs)`) works

Comment (jorisvandenbossche) [reply geopandas/geodataframe.py]: but you probably run into problems later on with eg `BlockManager(blocks, [columns, index])` where the index does not match the number of elements in the geometry column

**Comment by andreas-h** [comment]
If I don't do this, I get an `AttributeError: 'NoneType' object has no attribute 'columns'` at `if geometry in arg.columns:`

**Comment by andreas-h** [comment]
okay, now I'm also testing for the correct shape of the geodataframe.  works for me now.

**Comment by andreas-h** [comment]
mhh, i think the test is not good enough yet. will get back to that friday.

**Comment by mrocklin** [comment]
I may have inadvertently resolved this in https://github.com/geopandas/geopandas/pull/574 ?  It might be worth checking.

**Comment by jorisvandenbossche** [comment]
We are now going forward with PyGEOS (to be integrated in Shapely) instead of the cython branch for improving performance (see https://github.com/geopandas/geopandas/issues/1155, already available in master), and this issue is fixed and tested in master.

**Pull request closed without merging**